edb/server/args.py

```diff
@@ -94,6 +94,41 @@ class BackendCapabilitySets(NamedTuple):
     must_be_absent: List[pgsql_params.BackendCapabilities]
 
 
+class CompilerPoolScalingMode(enum.StrEnum):
+    Fixed = "fixed"
+    OnDemand = "on_demand"
+
+    def __init__(self, name):
+        self.pool_class = None
+
+    def set_class(self, cls):
```

Comment:
I think `assign_implementation` would be less confusing. Also add a comment that this is a decorator